lambda_rds.tf

```
@@ -0,0 +1,90 @@
+module "forwarder_rds_label" {
+  source     = "cloudposse/label/null"
+  version    = "0.24.1" # requires Terraform >= 0.13.0
+  attributes = ["forwarder-rds"]
+
+  context = module.this.context
+}
+
+module "forwarder_rds" {
+  count = local.lambda_enabled && var.forwarder_rds_enabled ? 1 : 0
+
+  source      = "cloudposse/module-artifact/external"
+  version     = "0.7.0"
+  filename    = "forwarder_rds.py"
+  module_name = var.dd_module_name
+  module_path = path.module
+  url         = "https://raw.githubusercontent.com/DataDog/datadog-serverless-functions/master/aws/rds_enhanced_monitoring/lambda_function.py?ref=${var.dd_forwarder_version}"
+}
+
+data "archive_file" "forwarder_rds" {
+  count       = local.lambda_enabled && var.forwarder_rds_enabled ? 1 : 0
+  type        = "zip"
+  source_file = module.forwarder_rds[0].file
+  output_path = "${path.module}/lambda.zip"
+}
+
+######################################################################
+## Create lambda function
+
+resource "aws_lambda_function" "forwarder_rds" {
+  count = local.lambda_enabled && var.forwarder_rds_enabled ? 1 : 0
+
+  #checkov:skip=BC_AWS_GENERAL_64: (Pertaining to Lambda DLQ) Vendor lambda does not have a means to reprocess failed events.
+
+  description                    = "Datadog forwarder for RDS enhanced monitoring."
+  filename                       = data.archive_file.forwarder_rds[0].output_path
+  function_name                  = module.lambda_label.id
+  role                           = aws_iam_role.lambda[0].arn
+  handler                        = "forwarder_rds.lambda_handler"
+  source_code_hash               = data.archive_file.forwarder_rds[0].output_base64sha256
+  runtime                        = var.lambda_runtime
+  reserved_concurrent_executions = var.lambda_reserved_concurrent_executions
+  tags                           = module.lambda_label.tags
+
+  dynamic "vpc_config" {
+    for_each = var.subnet_ids != null && var.security_group_ids != null ? [true] : []
```

```suggestion
    for_each = try(length(var.subnet_ids), 0) > 0 && try(length(var.security_group_ids), 0) > 0 ? [true] : []
```

let's check that those vars are 1) arrays; 2) not empty